bluemira/equilibria/run.py

```diff
@@ -218,73 +253,146 @@ def __init__(
+            warnings.warn(
+                warn_string.format("equilibrium_optimiser"),
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            self.eq_settings.opt_conditions = {
+                **(
+                    {}
+                    if equilibrium_optimiser.opt_conditions is None
+                    else equilibrium_optimiser.opt_conditions
+                ),
+                **self.eq_settings.opt_conditions,
+            }
+
+            self.eq_settings.algorithm = equilibrium_optimiser.algorithm_name
+            self.eq_settings.opt_parameters = {
+                **(
+                    {}
+                    if equilibrium_optimiser.opt_parameters is None
+                    else equilibrium_optimiser.opt_parameters
+                ),
+                **self.eq_settings.opt_parameters,
+            }
+
+        if equilibrium_convergence is not None:
+            warnings.warn(
+                warn_string.format("equilibrium_convergence"),
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            self.eq_settings.convergence = equilibrium_convergence
+        if equilibrium_problem_cls is not None:
+            warnings.warn(
+                warn_string.format("equilibrium_problem_cls"),
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            self.eq_settings.problem = equilibrium_problem_cls
+        # eos
 
         self._coil_cons = [] if coil_constraints is None else coil_constraints
         self.limiter = limiter
 
     @abstractmethod
     def optimise(self, *args, **kwargs) -> CoilSet:
-        """
-        Run pulsed coilset design optimisation
-        """
-        pass
+        """Run pulsed coilset design optimisation."""
+
+    @property
+    def bd_settings(self) -> BreakdownCOPSettings:
+        """Breakdown COP settings."""
+        return self.__bd_settings
 
     @property
     def _bd_settings(self) -> BreakdownCOPSettings:
         return self.__bd_settings
 
-    @_bd_settings.setter
-    def _bd_settings(self, value: Optional[Dict] = None):
-        self.__bd_settings = BreakdownCOPSettings(
-            **{
-                **{"B_stray_con_tol": 1e-8, "n_B_stray_points": 20},
-                **({} if value is None else value),
-            }
-        )
+    @bd_settings.setter
+    def bd_settings(self, value: Optional[Union[Dict, BreakdownCOPSettings]] = None):
+        """Breakdown COP settings."""
+        if value is None:
+            self.__bd_settings = BreakdownCOPSettings()
+        elif isinstance(value, BreakdownCOPSettings):
+            self.__bd_settings = value
+        else:
+            self.__bd_settings = BreakdownCOPSettings(**value)
+
+    @property
+    def eq_settings(self) -> EQSettings:
+        """Equilibrium COP settings."""
+        return self.__eq_settings
 
     @property
     def _eq_settings(self) -> EQSettings:
```

Comment:
Will this get dep'd infavour of just `eq_settings`?

Reply:
yep its already removed in #2320